server/game/cards/13.4-BtRK/TunnelsOfTheRedKeep.js

```diff
@@ -0,0 +1,23 @@
+const DrawCard = require('../../drawcard.js');
+
+class TunnelsOfTheRedKeep extends DrawCard {
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Kneel and return to shadows',
+            cost: ability.costs.kneelSelf(),
+            limit: ability.limit.perPhase(1),
+            handler: context => {
+                context.player.putIntoShadows(this, false);
+                this.game.addMessage('{0} kneels {1} to return it to shadows.', this.controller, this);
+                this.untilEndOfPhase(ability => ({
+                    match: card => card.getType() === 'character' && card.controller === this.controller,
+                    effect: ability.effects.dynamicStrength(() => this.controller.shadows.length)

```

Comment:
Everywhere in this handler should use `context.player` instead of `this.controller`. This is in case a player takes control of Tunnels, when the ability is activated and `putIntoShadows` finishes, it will go into its owner's shadow area, but the effect should apply to the player that stole and triggered it.